### PR TITLE
Made feedback redirection link & Accessibility Statement available in backend

### PR DESCRIPTION
**Pull request by Shubhamverma2796**
### Summary of Changes

Added the functionality for feedback redirection link & Accessibility statement. Site admins can select the Menu item where they want to redirect the user to report accessibility-related problems, and also for accessibility statements.

### Testing Instructions

Check that you can select the Menu Items for Accessibility statement and Accessibility Feeedback in the Accessibility plugin backend.

### Actual result BEFORE applying this Pull Request

Screenshot (435)

### Expected result AFTER applying this Pull Request

image

### CODE REVIEW

File: plugins/system/accessibility/accessibility.php [hunk 1]
```diff
@@ -13,6 +13,8 @@
 use Joomla\CMS\Factory;
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Router\Route;
+use Joomla\CMS\Language\Associations;
```
Comment (drmenzelit) [inline plugins/system/accessibility/accessibility.php]: Sort the use statements alphabetically

File: plugins/system/accessibility/accessibility.xml [hunk 2]
```diff
@@ -17,7 +17,7 @@
 		<language tag="en-GB">language/en-GB/plg_system_accessibility.sys.ini</language>
 	</languages>
 	<config>
-		<fields name="params">
+		<fields name="params" addfieldprefix="Joomla\Component\Menus\Administrator\Field">
```
Comment (drmenzelit) [inline plugins/system/accessibility/accessibility.xml]: I think the `addfieldprefix` should go inside the `modal_menu` field

File: plugins/system/accessibility/accessibility.php [hunk 3]
```diff
@@ -64,6 +82,12 @@ public function onBeforeCompileHead()
         // Detect the current active language
         $lang = Factory::getLanguage()->getTag();
 
+        // Get the item id of the menu item for feedback redirection
+        $menuItemId = $this->getAccessibilityItemId();
+
+        //generate the url to pass it to the accessibility script
+        $url = Route::link("site", "index.php?Itemid={$menuItemId}");
```
Comment (chmst) [inline plugins/system/accessibility/accessibility.php]: Use single quote .. compare similar code in joomla

File: administrator/language/en-GB/plg_system_accessibility.ini [hunk 4]
```diff
@@ -11,6 +11,8 @@ PLG_SYSTEM_ACCESSIBILITY_DECREASE_TEXT="Decrease Text Size"
 PLG_SYSTEM_ACCESSIBILITY_EMOJIS="Icons"
 PLG_SYSTEM_ACCESSIBILITY_EMOJIS_FALSE="Use Google Material Font"
 PLG_SYSTEM_ACCESSIBILITY_EMOJIS_TRUE="Use Emojis"
+PLG_SYSTEM_ACCESSIBILITY_FEEDBACK_REDIRECTION_LINK = "Feedback Redirection Link"
```
Comment (drmenzelit) [inline administrator/language/en-GB/plg_system_accessibility.ini]: Check blank spaces

File: administrator/language/en-GB/plg_system_accessibility.ini [hunk 5]
```diff
@@ -11,6 +11,8 @@ PLG_SYSTEM_ACCESSIBILITY_DECREASE_TEXT="Decrease Text Size"
...
+PLG_SYSTEM_ACCESSIBILITY_FEEDBACK_REDIRECTION_LINK_DESC = "User will be redirected to this link in case they want to give feedback regarding the accessibility of the website"
```
Comment (drmenzelit) [inline administrator/language/en-GB/plg_system_accessibility.ini]: Check blank spaces

File: plugins/system/accessibility/accessibility.xml (hunk 2 above)
Comment (Shubhamverma2796) [reply plugins/system/accessibility/accessibility.xml]: Done

File: plugins/system/accessibility/accessibility.php (hunk 3 above)
Comment (Shubhamverma2796) [reply plugins/system/accessibility/accessibility.php]: converted to single quote

File: administrator/language/en-GB/plg_system_accessibility.ini (hunk 5 above)
Comment (Shubhamverma2796) [reply administrator/language/en-GB/plg_system_accessibility.ini]: removed blank spaces

File: plugins/system/accessibility/accessibility.php (hunk 1 above)
Comment (Shubhamverma2796) [reply plugins/system/accessibility/accessibility.php]: sorted

Comment (drmenzelit) [reply plugins/system/accessibility/accessibility.php]: No

File: plugins/system/accessibility/accessibility.xml [hunk 6]
```diff
@@ -50,6 +50,7 @@
 					new="true"
 					edit="true"
 					clear="true"
+					addfieldprefix="Joomla\Component\Menus\Administrator\Field"
```
Comment (drmenzelit) [inline plugins/system/accessibility/accessibility.xml:58]: Put it as first element. Please check privacy consent plugin as example

File: plugins/system/accessibility/accessibility.php [hunk 7]
```diff
@@ -85,11 +101,13 @@ public function onBeforeCompileHead()
...
-        $menuItemId = $this->getAccessibilityItemId();
+        // Get the item id of the menu item for feedback redirection and accessibility statement
+        $feedbackMenuItemId = $this->getAccessibilityFeedbackItemId();
+        $statementMenuItemId = $this->getAccessibilityStatementItemId();
 
-        //generate the url to pass it to the accessibility script
-        $url = Route::link('site', 'index.php?Itemid={$menuItemId}');
+        //generate the urls to pass it to the accessibility script
+        $feedbackUrl = Route::link('site', 'index.php?Itemid='.$feedbackMenuItemId);
+        $statementUrl = Route::link('site', 'index.php?Itemid='.$statementMenuItemId);
```
Comment (chmst) [inline plugins/system/accessibility/accessibility.php]: Please add spaces before and after the concatenation operator.

**Pull request closed without merging**